shift_detector/checks/conditional_probabilities_check.py

```diff
@@ -20,60 +84,105 @@ class ConditionalProbabilitiesCheck(Check):
 
     def run(self, store):
-        rules, examined_columns = store[
-            ConditionalProbabilitiesPrecalculation(self.min_support, self.min_confidence)]
+        pre_calculation = store[
+            ConditionalProbabilitiesPrecalculation(self.min_support, self.min_confidence,
+                                                   self.number_of_bins, self.number_of_topics,
+                                                   self.min_delta_supports, self.min_delta_confidences)
+        ]
+        explanation = defaultdict(list)
 
-        filtered_rules = (rule for rule in rules if abs(rule.delta_supports) >= self.min_delta_supports and abs(
-            rule.delta_confidences) >= self.min_delta_confidences)
+        def add_to_explanation(rules, identifier):
+            for i, rule in enumerate(rules):
+                if i == self.rule_limit:
+                    break
+                explanation[identifier].append(rule)
 
-        compressed_rules = rule_compression.compress_and_sort_rules(filtered_rules)
+        add_to_explanation(pre_calculation.significant_rules_of_first, 'significant_rules_of_first')
+        add_to_explanation(pre_calculation.significant_rules_of_second, 'significant_rules_of_second')
+        add_to_explanation(pre_calculation.sorted_filtered_mutual_rules, 'sorted_filtered_mutual_rules')
 
-        shifted_columns = set()
-        explanation = defaultdict(list)
-        for i, compressed_rule in enumerate(compressed_rules):
-            if i == self.rule_limit:
-                break
-            columns = tuple(sorted(attribute for attribute, _ in compressed_rule.attribute_value_pairs))
-            shifted_columns.add(columns)
+        orange_rules_falling_below_min_delta_supports = sorted(
+            (rule for rule in pre_calculation.mutual_rules if abs(rule.delta_supports) < self.min_delta_supports
+             and abs(rule.delta_confidences) >= self.min_delta_confidences),
+            key=lambda r: (min(r.supports) != 0, abs(r.delta_confidences) * abs(r.delta_supports)),
+            reverse=True
+        )
+        add_to_explanation(orange_rules_falling_below_min_delta_supports,
+                           'orange_rules_falling_below_min_delta_supports')
+        orange_rules_falling_below_min_delta_confidences = sorted(
+            (rule for rule in pre_calculation.mutual_rules if
+             abs(rule.delta_confidences) < self.min_delta_confidences
+             and abs(rule.delta_supports) >= self.min_delta_supports),
+            key=lambda r: (min(r.supports) != 0, abs(r.delta_confidences) * abs(r.delta_supports)),
+            reverse=True
+        )
+        add_to_explanation(orange_rules_falling_below_min_delta_confidences,
+                           'orange_rules_falling_below_min_delta_confidences')
 
-            explanation[', '.join(columns)].append(str(compressed_rule))
+        shifted_columns = set(tuple(sorted(attribute for attribute, _ in rule.left_side + rule.right_side))
```

Comment:
You don't need to convert this into a tuple if you convert into a set right after. just turn it into a set directly

Reply:
It's not a tuple, it's an iterator producing tuples.